### PR TITLE
#175636836 Integration of coveralls code coverage

### PR DESCRIPTION
**Pull request by fabriceumuhire**
#### What does this PR do?
- Integrate coveralls and generate code coverage report 
#### Description of Task to be completed?
-  run `npm install coveralls --save-dev` to install coveralls dependencies package
- run `npm i -D nyc` to install instanbul dependencies package
- add `"coverage": "nyc report --reporter=text-lcov | coveralls"` in package.json file
- Sync `tech-strikers-bn-backend` repository to coveralls
- Add badge to README.md file
#### How should this be manually tested?
- run `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- run `cd tech-strikers-bn-backend`
- run `npm run coverage`
#### What are the relevant pivotal tracker stories?
- [#175636836](https://www.pivotaltracker.com/story/show/175636836)
#### Screenshots (if appropriate)
- Coveralls setup
image

### CODE REVIEW

**Review by Duncanwar — APPROVED** [review]
It is ok.

**Comment by bigelio** [comment]
It's ok but it would be better to provide background context on unit testing and move coverall package in dependancies instead of devDependencies

File: .gitignore [hunk 1]
```diff
@@ -46,6 +46,10 @@ build/Release
 node_modules/
 jspm_packages/
 
+# package-lock
+
+package-lock.json
```
Comment (kipropbrian) [inline .gitignore]: Remove package.lock

Comment (fabriceumuhire) [reply .gitignore]: Done!

**Comment by fabriceumuhire** [comment]
> It's ok but it would be better to provide background context on unit testing and move coverall package in dependancies instead of devDependencies

Done!

**Review by geraudirad — APPROVED** [review]
Good work on this Pull request

**Pull request merged**